Model/Button/GetCartButtonConfig.php

```diff
@@ -35,11 +37,7 @@ public function execute(
         $quote = null;
 
         if ($cartId) {
-            $quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
-            if ($quoteIdMask) {
-                $quoteId = (int) $quoteIdMask->getQuoteId();
-                $quote = $this->cartRepository->get($quoteId);
-            }
+            $quote = $this->guestCartRepository->get($cartId);
```

Comment:
The logic in guestCartRepository is:

```php
$quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
return $this->quoteRepository->get($quoteIdMask->getQuoteId());
```

Which is very similar to what we already did, and might not be the root of the problem we are trying to fix.

What we could do is implement `\Magento\Quote\Model\MaskedQuoteIdToQuoteIdInterface::execute` instead to simplify the logic here and be more robust